### ISSUE TITLE
How might I use DPO for my custom model?

### ISSUE THREAD

**Issue by Tony363** [eric-mitchell/direct-preference-optimization#7]
Hello,

I have finetuned a preferred Vision Language model for VQA task. How use FSDPTrainer to do DPO for my preference model and policy model of the same architecture?

Tony,

**Comment by eric-mitchell** [comment]
Hi Tony. Not sure what you mean by "preferred vision language model", but I'll assume that you're talking about some model that maps images to text.

DPO is architecture-agnostic, so in principle there's no reason you couldn't fine-tune a vision-language model with it. To use this codebase, you'd need to modify the FSDPTrainer a bit, though, since we currently only support language models out of the box.
Off the top of my head, you'll need to update `FSDPTrainer` to:
- do FSDP wrapping of your VLM correctly, which might be a bit different than a language-only transformer
- change `get_batch_metrics` to use your image inputs correctly to produce logprobs of your text labels for both policy and reference model, conditioned on the image
and modify `preference_datasets.py` to load your vision-language dataset.

**Comment by Tony363** [comment]
My apologies, please disregard the "preferred" word. Ill try out your suggestions!

**Comment by eric-mitchell** [comment]
Got it- let me know if you're having any issues, and I will try my best to help. Would be very interested to see what DPO can do for your VLM!
